### PR TITLE
Fix TFElectraForMultipleChoice

### PR DESCRIPTION
**Pull request by ydshieh**
# What does this PR do?

Current `TFElectraForMultipleChoice` calls `TFElectraMainLayer` using positional arguments, and it doesn't have

```
        encoder_hidden_states,
        encoder_attention_mask
        past_key_values
        use_cache
```
so `inputs["output_attentions"]` is received as `encoder_hidden_states` by the main layer, etc.

This PR fixes it (using kwargs).

@Rocketknight1 @gante

### CODE REVIEW

**Review by gante — APPROVED** [review]
👍  ~It shouldn't change any results, but it is more readable~

@sgugger doublechecking if I can merge

**Comment by ydshieh** [comment]
Hi, @gante . It does affect the results - I made a typo in the original message. 

```
so inputs["output_attentions"] is received as encoder_hidden_states by the main layer, etc.
```
(and this is why a correction is needed.)

**Comment by gante** [comment]
You're absolutely right, it was being passed to the wrong variables. Thanks for lending your keen eye 👁️

**Comment by gante** [comment]
This also means that `TFElectraForMultipleChoice` was not being properly tested -- how did tests like `test_pt_tf_model_equivalence`, which require `output_hidden_states = True` across the layers, were passing? (I've checked the [PT implementation](https://github.com/huggingface/transformers/blob/master/src/transformers/models/electra/modeling_electra.py#L1477), it is passing the keyword arguments correctly)

**Comment by ydshieh** [comment]
> This also means that `TFElectraForMultipleChoice` was not being properly tested -- how did tests like `test_pt_tf_model_equivalence`, which require `output_hidden_states = True` across the layers, were passing? (I've checked the [PT implementation](https://github.com/huggingface/transformers/blob/master/src/transformers/models/electra/modeling_electra.py#L1477), it is passing the keyword arguments correctly, so this mismatch should have been caught)

Current TF Electra Test's `prepare_config_and_inputs_for_common` only provides `inputs_dict = {"input_ids": input_ids, "token_type_ids": token_type_ids, "attention_mask": input_mask}`. So no `output_hidden_states` is used for the test.

In the enhanced `test_pt_tf_model_equivalence`, I manually set `output_hidden_states=True` and `output_attentions` in order to have a more complete check --> and this issue is then identified.

**Comment by gante** [comment]
The equivalence test sets the `output_hidden_states` to `True` ([here](https://github.com/huggingface/transformers/blob/master/tests/test_modeling_tf_common.py#L355)), and on the line that you've changed this PR, I can confirm that `inputs["output_hidden_states"]` is `True` during the test :)

However, because it was set to `True` in the config, it may overwrite the `None` in the `ElectraMainLayer`. In other words, before this PR, `output_hidden_states` was not being set (because of the error you uncovered 🙌 ), so the config value took charge, and the test passed anyway.

In other words, we probably should be more strict about the use of keyword arguments, to avoid these sorts of errors -- especially because the config values may override a `None` :)

**Comment by ydshieh** [comment]
Great analysis about `output_hidden_states` , @gante , you are right. 

However, the equivalence test only tests the first output:

https://github.com/huggingface/transformers/blob/525dbbf84a0d2933686281c513689da9794b7dd1/tests/test_modeling_tf_common.py#L384

I think the test will pass anyway even if there is a **actual** mismatch (i.e. even if the config doesn't take the chager).

----

I was thinking more why `encoder_hidden_states` gets value `False` and `encoder_attention_mask` gets `True` don't cause any problem - it turns out that these values are used only when the model is used as `decoder`, and `TFElectraForMultipleChoice` is not (so these wrong values are just ignored by the model without causing any exception)

----

**Review by sgugger — APPROVED** [review]
Thanks for fixing!

**Comment by gante** [comment]
@ydshieh Keep these fantastic PRs coming 🙏
